src/components/ScrollButton.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+
+import { useState } from "react";
+
+import styles from "./ScrollButton.module.css";
+
+export function ScrollButton() {
+	const [visible, setVisible] = useState(false);
+
+	const toggleVisible = () => {
+		const scrolled = document.body.scrollTop;
+		setVisible(scrolled > 100);
+	};
+
+	const scrollToTop = () => {
+		document.body.scrollTo({
+			behavior: "smooth",
+			top: 0,
+		});
+	};
+
+	document.body.addEventListener("scroll", toggleVisible, { passive: true });
```

Comment:
```plaintext
- error src/components/ScrollButton.tsx (22:1) @ document
- error ReferenceError: document is not defined
    at ScrollButton (./src/components/ScrollButton.tsx:26:5)
  20 |  };
  21 | 
> 22 |  document.body.addEventListener("scroll", toggleVisible, { passive: true });
     | ^
  23 | 
  24 |  return (
  25 |          <button
```

Reply:
https://dev.to/vvo/how-to-solve-window-is-not-defined-errors-in-react-and-next-js-5f97